### PR TITLE
[Core][MeshingApplication] Extend gradient process to other geometries. Reduce the number of template arguments to reduce compilation time

### PR DESCRIPTION
**Pull request by loumalouomega**
This allows to use the gradient an Hessian with geometries different than triangles and tetras. Additionally it reduces significantly the number of template arguments required

@RiccardoTosi I will add the corrections you suggested on the Hessian in a future PR

### CODE REVIEW

File: kratos/tests/cpp_tests/processes/test_compute_nodal_gradient_process.cpp [hunk 1]
```diff
@@ -258,22 +268,183 @@ namespace Kratos
...
+            // First we create the nodes 
+            NodeType::Pointer p_node_1 = this_model_part.CreateNewNode(1, 0.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_2 = this_model_part.CreateNewNode(2, 1.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_3 = this_model_part.CreateNewNode(3, 1.0 , 1.0 , 0.0);
+            NodeType::Pointer p_node_4 = this_model_part.CreateNewNode(4, 0.0 , 1.0 , 0.0);
+            NodeType::Pointer p_node_5 = this_model_part.CreateNewNode(5, 2.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_6 = this_model_part.CreateNewNode(6, 2.0 , 1.0 , 0.0);
+            
+            // Initialize nodal area
+            for (auto& node : this_model_part.Nodes())
```
Comment (philbucher) [inline kratos/tests/cpp_tests/processes/test_compute_nodal_gradient_process.cpp:330]: variable-utils?

File: kratos/python/add_processes_to_python.cpp [hunk 2]
```diff
@@ -274,42 +274,26 @@ void  AddProcessesToPython(pybind11::module& m)
...
+    .def(py::init<ModelPart&, Variable<double>&, Variable<array_1d<double,3> >& , Variable<double>& >())
     ;
 
-    py::class_<ComputeNodalGradientProcess<3, component_type, NonHistorical>, ComputeNodalGradientProcess<3, component_type, NonHistorical>::Pointer, Process>(m,"ComputeNonHistoricalNodalGradientProcessComp3D")
-            .def(py::init<ModelPart&, component_type&, Variable<array_1d<double,3> >& , Variable<double>& >())
-    ;
+    m.attr("ComputeNonHistoricalNodalGradientProcess2D") = m.attr("ComputeNonHistoricalNodalGradientProcess");
+    m.attr("ComputeNonHistoricalNodalGradientProcess3D") = m.attr("ComputeNonHistoricalNodalGradientProcess");
+    m.attr("ComputeNonHistoricalNodalGradientProcessComp2D") = m.attr("ComputeNonHistoricalNodalGradientProcess");
+    m.attr("ComputeNonHistoricalNodalGradientProcessComp3D") = m.attr("ComputeNonHistoricalNodalGradientProcess");
```
Comment (philbucher) [inline kratos/python/add_processes_to_python.cpp:302]: what are you doing here?

File: applications/MeshingApplication/tests/mmg_lagrangian_test/beam3D_test_result.sol [hunk 3]
```diff
@@ -7,173 +7,173 @@ Dimension
...
-7.87642627781283  0  7.87642627781283  0  0  7.87642627781283  
-8.33949942560789  0  8.33949942560789  0  0  8.33949942560789  
-8.5807679085591  0  8.5807679085591  0  0  8.5807679085591  
-7.85107300473627  0  7.85107300473627  0  0  7.85107300473627  
-8.42153377613511  0  8.42153377613511  0  0  8.42153377613511  
-6.91498232488896  0  6.91498232488896  0  0  6.91498232488896  
-8.02491969678893  0  8.02491969678893  0  0  8.02491969678893  
-7.40582739573172  0  7.40582739573172  0  0  7.40582739573172  
-7.15130340264934  0  7.15130340264934  0  0  7.15130340264934  
+27.7057416888572  0  27.7057416888572  0  0  27.7057416888572
```
Comment (philbucher) [inline applications/MeshingApplication/tests/mmg_lagrangian_test/beam3D_test_result.sol:10]: why do the results change?
I thought this is only refactoring?

File: applications/MeshingApplication/tests/mmg_lagrangian_test/beam2D_test_result.sol [hunk 4]
```diff
@@ -7,77 +7,77 @@ Dimension
 SolAtVertices
 205
 1 3
- 35.9101637379947 0 35.9101637379947
- 37.4367319907306 0 37.4367319907306
- 26.2566573854866 0 26.2566573854866
- 33.410918127346 0 33.410918127346
+ 35.9143759191543 0 35.9143759191543
```
Comment (philbucher) [inline applications/MeshingApplication/tests/mmg_lagrangian_test/beam2D_test_result.sol]: same

File: applications/MeshingApplication/tests/cpp_tests/test_metric_process.cpp [hunk 5]
```diff
@@ -303,11 +304,12 @@ namespace Kratos
                 auto it_node = this_model_part.Nodes().begin() + i_node;
                 it_node->FastGetSolutionStepValue(DISTANCE) = (it_node->X() == 1.0) ? 0.0 : 1.0;
                 it_node->SetValue(NODAL_H, 1.0);
+                it_node->SetValue(NODAL_AREA, 0.0);
                 it_node->SetValue(METRIC_TENSOR_3D, ZeroVector(6));
             }
 
             // Compute gradient
-            typedef ComputeNodalGradientProcess<3, Variable<double>, Historical> GradientType;
+            typedef ComputeNodalGradientProcess<ComputeNodalGradientProcessSettings::SaveAsHistoricalVariable> GradientType;
```
Comment (philbucher) [inline applications/MeshingApplication/tests/cpp_tests/test_metric_process.cpp:314]: why not one test with historical and the other with non-historical?

File: kratos/processes/compute_nodal_gradient_process.cpp [hunk 6]
```diff
@@ -26,42 +23,75 @@
...
-        GeometryUtils::CalculateGeometryData(geom, DN_DX, N, Volume);
-        
-        array_1d<double, TDim+1> values;
-        for(std::size_t i=0; i<TDim+1; ++i)
-            values[i] = geom[i].FastGetSolutionStepValue(mrOriginVariable);
+    const auto& it_element_begin = mrModelPart.ElementsBegin();
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
+    const std::size_t local_space_dimension = r_first_element_geometry.LocalSpaceDimension();
+    const std::size_t number_of_nodes = r_first_element_geometry.PointsNumber();
```
Comment (philbucher) [inline kratos/processes/compute_nodal_gradient_process.cpp]: same comment as in the other pr, waht happens if different elements are in a modelpart?

File: kratos/processes/compute_nodal_gradient_process.cpp [hunk 7]
```diff
@@ -26,42 +23,75 @@
...
+            MathUtils<double>::InvertMatrix(J0, InvJ0, detJ0);
+            const Matrix& rDN_De = rDN_DeContainer[point_number];
+            GeometryUtils::ShapeFunctionsGradients(rDN_De, InvJ0, DN_DX);
+            
+            const Vector grad = prod(trans(DN_DX), values);
+            const double gauss_point_volume = integration_points[point_number].Weight() * detJ0;
+            
+            for(std::size_t i_node=0; i_node<number_of_nodes; ++i_node) {
+                for(std::size_t k=0; k<dimension; ++k) {
+                    double& val = GetGradient(r_geometry, i_node,k);
```
Comment (philbucher) [inline kratos/processes/compute_nodal_gradient_process.cpp]: this algorithm looks familiar, didn't you recently implement sth similar?

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 8]
```diff
@@ -20,148 +20,84 @@
...
     // Some checks
-    VariableUtils().CheckVariableExists(mVariable, nodes_array);
+    NodesArrayType& nodes_array = mThisModelPart.Nodes();
+    if (mrOriginVariableDoubleList.size() > 0) {
+        VariableUtils().CheckVariableExists(mrOriginVariableDoubleList[0], nodes_array);
+    } else {
+        VariableUtils().CheckVariableExists(mrOriginVariableComponentsList[0], nodes_array);
+    }
     for (auto& i_node : nodes_array)
         KRATOS_ERROR_IF_NOT(i_node.Has(NODAL_H)) << "NODAL_H must be computed" << std::endl;
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:74]:
```suggestion
         KRATOS_ERROR_IF_NOT(i_node.Has(NODAL_H)) << "NODAL_H must be computed, node-Id: " << i_node.Id() << std::endl;
```

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 9]
```diff
@@ -20,148 +20,84 @@
...
 
     // Some checks
-    VariableUtils().CheckVariableExists(mVariable, nodes_array);
+    NodesArrayType& nodes_array = mThisModelPart.Nodes();
+    if (mrOriginVariableDoubleList.size() > 0) {
+        VariableUtils().CheckVariableExists(mrOriginVariableDoubleList[0], nodes_array);
+    } else {
+        VariableUtils().CheckVariableExists(mrOriginVariableComponentsList[0], nodes_array);
+    }
     for (auto& i_node : nodes_array)
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]:
```suggestion
     for (const auto& i_node : nodes_array)
```

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 10]
```diff
@@ -20,148 +20,84 @@
...
-    NodesArrayType& nodes_array = mThisModelPart.Nodes();
-    const int num_nodes = nodes_array.end() - nodes_array.begin();
-
     CalculateAuxiliarHessian();
 
     // Some checks
-    VariableUtils().CheckVariableExists(mVariable, nodes_array);
+    NodesArrayType& nodes_array = mThisModelPart.Nodes();
+    if (mrOriginVariableDoubleList.size() > 0) {
+        VariableUtils().CheckVariableExists(mrOriginVariableDoubleList[0], nodes_array);
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: side comment, this function does not take its args as `const`!

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 11]
```diff
@@ -20,148 +20,84 @@
...
-        if (it_node->SolutionStepsDataHas(reference_var)) {
-            const double ratio_reference = it_node->FastGetSolutionStepValue(reference_var);
-            ratio = CalculateAnisotropicRatio(ratio_reference, mAnisotropicRatio, mBoundLayer, mInterpolation);
-        }
-
-        // For postprocess pourposes
-        it_node->SetValue(ANISOTROPIC_RATIO, ratio);
+    const auto& it_element_begin = mThisModelPart.ElementsBegin();
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: I would say the processinfo is safer

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 12]
```diff
@@ -219,103 +155,153 @@ array_1d<double, 3 * (TDim - 1)> ComputeHessianSolMetricProcess<TDim, TVarType>:
...
+    const auto& it_element_begin = elements_array.begin();
 
-    #pragma omp parallel for
-    for(int i = 0; i < num_nodes; ++i)
-        (nodes_array.begin() + i)->SetValue(AUXILIAR_HESSIAN, aux_zero_vector);
+    // Geometry information
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
+    const std::size_t local_space_dimension = r_first_element_geometry.LocalSpaceDimension();
+    const std::size_t number_of_nodes = r_first_element_geometry.PointsNumber();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: same comment as before

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 13]
```diff
@@ -343,9 +329,135 @@ double ComputeHessianSolMetricProcess<TDim, TVarType>::CalculateAnisotropicRatio
...
+Parameters ComputeHessianSolMetricProcess::GetDefaultParameters()
+{
+    Parameters default_parameters = Parameters(R"(
+    {
+        "minimal_size"                        : 0.1,
+        "maximal_size"                        : 10.0,
+        "enforce_current"                     : true,
+        "hessian_strategy_parameters":
+        {
+            "metric_variable"                  : ["DISTANCE"],
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:428]:
```suggestion
             "metric_variables"                  : ["DISTANCE"],
```
it is a list

**Review by philbucher — COMMENTED** [review]
wasn't it you that introduced the template-args?

File: kratos/tests/cpp_tests/processes/test_compute_nodal_gradient_process.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/tests/cpp_tests/processes/test_compute_nodal_gradient_process.cpp:330]: It is a test. The test already take long to compile to add more dependencies

File: kratos/python/add_processes_to_python.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/python/add_processes_to_python.cpp:302]: It is to for retro-compatibility reasons, this way it is compiled only once, and some alias are created

File: applications/MeshingApplication/tests/mmg_lagrangian_test/beam3D_test_result.sol (hunk 3 above)
Comment (loumalouomega) [reply applications/MeshingApplication/tests/mmg_lagrangian_test/beam3D_test_result.sol:10]: If you take a look deeper will see that the values are reordered. I can control the output of the sol files of MMG

File: applications/MeshingApplication/tests/mmg_lagrangian_test/beam2D_test_result.sol (hunk 4 above)
Comment (loumalouomega) [reply applications/MeshingApplication/tests/mmg_lagrangian_test/beam2D_test_result.sol]: Same

File: applications/MeshingApplication/tests/cpp_tests/test_metric_process.cpp (hunk 5 above)
Comment (loumalouomega) [reply applications/MeshingApplication/tests/cpp_tests/test_metric_process.cpp:314]: Right now the level set process only considers historical values

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 13 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:428]: I will keep it. It is used in many places, and will break the cod of the current users as @AlejandroCornejo

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 11 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: For what?

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 10 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: CheckVariableExists?

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 12 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: I don't get what you mean

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 10 above)
Comment (philbucher) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: yes

**Comment by philbucher** [comment]
why do you make your algorithm to only consider the first element to determine the number of node, dimension?
I had the same comment in another PR of you some days ago

This can be easily broken!

**Comment by loumalouomega** [comment]
Yes, you are right. This branch was created at the same time and I had not
though about the repeated pieces of code. I will change that.

**Comment by loumalouomega** [comment]
Done @philbucher

File: kratos/processes/compute_nodal_gradient_process.cpp [hunk 14]
```diff
@@ -26,42 +23,89 @@
...
-    for(int i=0; i<static_cast<int>(mrModelPart.Elements().size()); ++i) {
-        auto it_elem = mrModelPart.ElementsBegin()+i;
-        Element::GeometryType& geom = it_elem->GetGeometry();
-        GeometryUtils::CalculateGeometryData(geom, DN_DX, N, Volume);
+    // First element iterator
+    const auto it_element_begin = mrModelPart.ElementsBegin();
+    
+    // Geometry information
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
```
Comment (philbucher) [inline kratos/processes/compute_nodal_gradient_process.cpp]:
```suggestion
     const std::size_t dimension = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
```

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 15]
```diff
@@ -343,9 +349,142 @@ double ComputeHessianSolMetricProcess<TDim, TVarType>::CalculateAnisotropicRatio
...
+        } else {
+            metric = ComputeHessianMetricTensor<TDim>(hessian, ratio, element_min_size, element_max_size, nodal_h);
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Parameters ComputeHessianSolMetricProcess::GetDefaultParameters()
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: const

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 16]
```diff
@@ -219,103 +165,163 @@ array_1d<double, 3 * (TDim - 1)> ComputeHessianSolMetricProcess<TDim, TVarType>:
...
+    ElementsArrayType& elements_array = mThisModelPart.Elements();
+    const int num_elements = static_cast<int>(elements_array.size());
+    const auto& it_element_begin = elements_array.begin();
 
-    #pragma omp parallel for
-    for(int i = 0; i < num_nodes; ++i)
-        (nodes_array.begin() + i)->SetValue(AUXILIAR_HESSIAN, aux_zero_vector);
+    // Geometry information
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]:
```suggestion
     const std::size_t dimension = mThisModelPart.GetProcessInfo()[DOMAIN_SIZE];
```

**Review by philbucher — COMMENTED** [review]
Take a look at codacy (minor)

If you use DOMAIN_SIZE to determine the dimension the you could add a check if the value is reasonable

**Comment by loumalouomega** [comment]
@philbucher Comments adressed

File: kratos/processes/compute_nodal_gradient_process.cpp (hunk 7 above)
Comment (RiccardoRossi) [reply kratos/processes/compute_nodal_gradient_process.cpp]: i  think i commented this algorithm to Daniel Diez in his PR on distance calculation

Comment (RiccardoRossi) [reply kratos/processes/compute_nodal_gradient_process.cpp]: i  think i commented this algorithm to Daniel Diez in his PR on distance calculation

HOWEVER on line 84 you access many times to the database. that's somewhat slow...and you coukd at least avoid doing it once per component

**Comment by loumalouomega** [comment]
Ping @philbucher

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 11 above)
Comment (philbucher) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: determining the dimension of the problem

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 8 above)
Comment (philbucher) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:74]: ping

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 17]
```diff
@@ -20,157 +20,103 @@
...
+    ModelPart& rThisModelPart,
+    ComponentType& rVariable,
+    Parameters ThisParameters
+    ):mThisModelPart(rThisModelPart)
+{
+    // We push the components list
+    mrOriginVariableComponentsList.push_back(&rVariable);
+
+    // We check the parameters
+    Parameters default_parameters = GetDefaultParameters();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:51]: const

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 18]
```diff
@@ -20,157 +20,103 @@
...
-            "hmin_over_hmax_anisotropic_ratio"     : 1.0,
-            "boundary_layer_max_distance"          : 1.0,
-            "interpolation"                        : "Linear"
-        }
-    })" );
+    // We push the list of double variables
+    mrOriginVariableDoubleList.push_back(&rVariable);
 
+    // We check the parameters
+    Parameters default_parameters = GetDefaultParameters();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:33]: const

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 19]
```diff
@@ -20,157 +20,103 @@
...
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
 
-            metric = MetricsMathUtils<TDim>::IntersectMetrics(old_metric, new_metric);
-        } else {
-            metric = ComputeHessianMetricTensor(hessian, ratio, element_min_size, element_max_size);
-        }
+    if (dimension == 2) {
+        CalculateMetric<2>();
+    } else {
+        CalculateMetric<3>();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:83]: else if dim == 3

else: Error

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 15 above)
Comment (philbucher) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: ping

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 20]
```diff
@@ -343,9 +348,142 @@ double ComputeHessianSolMetricProcess<TDim, TVarType>::CalculateAnisotropicRatio
...
+        } else {
+            metric = ComputeHessianMetricTensor<TDim>(hessian, ratio, element_min_size, element_max_size, nodal_h);
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Parameters ComputeHessianSolMetricProcess::GetDefaultParameters()
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]:
```suggestion
Parameters ComputeHessianSolMetricProcess::GetDefaultParameters() const
```

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 21]
```diff
@@ -343,9 +348,142 @@ double ComputeHessianSolMetricProcess<TDim, TVarType>::CalculateAnisotropicRatio
...
+    return default_parameters;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ComputeHessianSolMetricProcess::InitializeVariables(Parameters ThisParameters)
+{
+    // Get default variables
+    Parameters default_parameters = GetDefaultParameters();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:464]: const

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 22]
```diff
@@ -343,9 +348,142 @@ double ComputeHessianSolMetricProcess<TDim, TVarType>::CalculateAnisotropicRatio
...
+
+    // Set variables
+    mMinSize = ThisParameters["minimal_size"].GetDouble();
+    mMaxSize = ThisParameters["maximal_size"].GetDouble();
+    mEnforceCurrent = ThisParameters["enforce_current"].GetBool();
```
Comment (philbucher) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:469]: all const

**Review by philbucher — COMMENTED** [review]
some comments are not addressed

plus some more minor comments

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 18 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:33]: RecursivelyValidateAndAssignDefaults can not work with const Parameters

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 22 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:469]: Member variables, cannot be const

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 21 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:464]: Cannot be const, (the function can be), but not the parameters. Maybe adding a new method to the Parameters (the one is still in the oldest open PR)

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 18 above)
Comment (philbucher) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:33]: indeed, I checked, even though I think it should be 
=> maybe after #58

**Review by philbucher — APPROVED** [review]
looks good to me now :+1:

**Pull request merged**